src/main/java/org/dependencytrack/tasks/scanners/SnykAnalysisTask.java

```diff
@@ -0,0 +1,459 @@
+                            }
+                            Event.dispatch(new IndexEvent(IndexEvent.Action.COMMIT, Vulnerability.class));
+                            updateAnalysisCacheStats(qm, Vulnerability.Source.SNYK, API_BASE_URL, component.getPurl().toString(), component.getCacheResult());
+                        }
+                    }
+                }
+            } else if (responseCode == 404) {
+                Vulnerability vulnerability = new Vulnerability();
+                addVulnerabilityToCache(component, vulnerability);
+                updateAnalysisCacheStats(qm, Vulnerability.Source.SNYK, API_BASE_URL, component.getPurl().toString(), component.getCacheResult());
+                handleUnexpectedHttpResponse(LOGGER, API_BASE_URL, responseCode, statusText);
+            }
+        }
+    }
+
+    public JSONObject selectCvssObjectBasedOnSource(JSONArray cvssArray) {
+
+        String cvssSourceHigh = getSnykCvssConfig(ConfigPropertyConstants.SCANNER_SNYK_CVSS_SOURCE);
+        String cvssSourceLow = cvssSourceHigh.equalsIgnoreCase(SnykCvssSource.NVD.toString()) ?
+                SnykCvssSource.SNYK.toString() :
+                SnykCvssSource.NVD.toString();
+        JSONObject cvss = cvssArray.optJSONObject(0);
+        if (cvssArray.length() > 1) {
+            for (int i = 0; i < cvssArray.length(); i++) {
+                final JSONObject cvssObject = cvssArray.optJSONObject(i);
+                String source = cvssObject.optString("source");
+                String vector = cvssObject.optString("vector");
+                String score = cvssObject.optString("score");
+                if (!source.isBlank() && !vector.isBlank() && !score.isBlank()) {
+                    if (source.equalsIgnoreCase(cvssSourceHigh)) {
+                        return cvssObject;
+                    }
+                    if (source.equalsIgnoreCase(cvssSourceLow)) {
+                        cvss = cvssObject;
+                    } else {
+                        if (cvss != null && !cvss.optString("source").equalsIgnoreCase(cvssSourceLow)) {
+                            cvss = cvssObject;
+                        }
+                    }
+                }
+            }
+        }
+        return cvss;
+    }
+
+    public String getSnykCvssConfig(ConfigPropertyConstants scannerSnykCvssSource) {
+
+        try (QueryManager qm = new QueryManager()) {
+            final ConfigProperty property = qm.getConfigProperty(
+                    scannerSnykCvssSource.getGroupName(), scannerSnykCvssSource.getPropertyName()
+            );
+            if (property != null && ConfigProperty.PropertyType.STRING == property.getPropertyType()) {
+                return property.getPropertyValue();
+            }
+        }
+        return scannerSnykCvssSource.getDefaultPropertyValue();
+    }
+
+    public List<VulnerableSoftware> parseVersionRanges(final QueryManager qm, final String purl, final JSONArray ranges) {
```

Comment:
If we mirror the affected ranges, the next time a component matching this criteria will be scanned, it will be picked up by the internal analyzer, not the Snyk one. This will also cause the attribution in the UI to show as "INTERNAL", not "SNYK". 

For now, I'd prefer it if we leave the mirroring of `VulnerableSoftware` out, and just rely on using the API and caching.